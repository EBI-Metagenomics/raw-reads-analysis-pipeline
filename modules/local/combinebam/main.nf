process COMBINEBAM {
    tag "${meta.id}"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'https://depot.galaxyproject.org/singularity/samtools:1.21--h50ea8bc_0'
        : 'quay.io/biocontainers/samtools:1.21--h50ea8bc_0'}"

    input:
    tuple val(meta), path(input_files, stageAs: "inputs/?/*")

    output:
    tuple val(meta), path("${output_name}"), emit: concatenated_result
    path "versions.yml"                    , emit: versions

    script:
    def args = task.ext.args ?: ''
    output_name = (input_files instanceof List ? input_files[0] : input_files).name.tokenize('/')[-1]

    """
    samtools cat -o ${output_name} \$(ls inputs/*/*) ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        combinebam: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    output_name = input_files[0].name

    """
    touch ${output_name}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        combinebam: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
    END_VERSIONS
    """
}
