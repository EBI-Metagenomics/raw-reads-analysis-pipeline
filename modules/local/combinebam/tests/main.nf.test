nextflow_process {

    name "Test Process COMBINEBAM"
    script "../main.nf"
    process "COMBINEBAM"

    tag "modules"
    tag "modules_"
    tag "combinebam"

    test("one BAM should match snapshot") {

        when {
            process {
                """
                input[0] = [
		    ['id': 'test'],
		    [
		        file("${projectDir}/tests/fixtures/combinebam/ERR10889188_mini.bam", checkIfExists: true),
		    ]
		]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("two BAMs should match snapshot") {

        when {
            process {
                """
                input[0] = [
		    ['id': 'test'],
		    [
		        file("${projectDir}/tests/fixtures/combinebam/ERR10889188_mini.bam", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/combinebam/ERR10889188_mini.bam", checkIfExists: true),
		    ]
		]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("10 BAMs should match snapshot") {

        when {
            process {
                """
                input[0] = [
		    ['id': 'test'],
		    [
		        file("${projectDir}/tests/fixtures/combinebam/ERR10889188_mini.bam", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/combinebam/ERR10889188_mini.bam", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/combinebam/ERR10889188_mini.bam", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/combinebam/ERR10889188_mini.bam", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/combinebam/ERR10889188_mini.bam", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/combinebam/ERR10889188_mini.bam", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/combinebam/ERR10889188_mini.bam", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/combinebam/ERR10889188_mini.bam", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/combinebam/ERR10889188_mini.bam", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/combinebam/ERR10889188_mini.bam", checkIfExists: true),
		    ]
		]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
}
