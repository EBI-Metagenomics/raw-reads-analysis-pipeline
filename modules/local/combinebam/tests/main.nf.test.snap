{
    "10 BAMs should match snapshot": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "ERR10889188_mini.bam:md5,080e32ac1e5b403e5dac7c1bd0f6263f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4ec4b3de4a6329dd6acc3e3f4ff6f815"
                ],
                "concatenated_result": [
                    [
                        {
                            "id": "test"
                        },
                        "ERR10889188_mini.bam:md5,080e32ac1e5b403e5dac7c1bd0f6263f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4ec4b3de4a6329dd6acc3e3f4ff6f815"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-05-20T13:52:44.711904316"
    },
    "two BAMs should match snapshot": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "ERR10889188_mini.bam:md5,8f9919214c80a9656959862891e12b7b"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4ec4b3de4a6329dd6acc3e3f4ff6f815"
                ],
                "concatenated_result": [
                    [
                        {
                            "id": "test"
                        },
                        "ERR10889188_mini.bam:md5,8f9919214c80a9656959862891e12b7b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4ec4b3de4a6329dd6acc3e3f4ff6f815"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-05-20T13:52:23.280267236"
    },
    "one BAM should match snapshot": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "ERR10889188_mini.bam:md5,78d513132a0c933891c9bbf194b7ce34"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4ec4b3de4a6329dd6acc3e3f4ff6f815"
                ],
                "concatenated_result": [
                    [
                        {
                            "id": "test"
                        },
                        "ERR10889188_mini.bam:md5,78d513132a0c933891c9bbf194b7ce34"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4ec4b3de4a6329dd6acc3e3f4ff6f815"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-05-20T13:51:55.320399855"
    }
}